pkg/cli/gettoken/gettoken.go

```diff
@@ -0,0 +1,269 @@
+package gettoken
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/browser"
+	"github.com/spf13/cobra"
+	"golang.org/x/sync/errgroup"
+
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/client-go/util/homedir"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	"github.com/openshift/oc/pkg/cli/gettoken/credwriter"
+	"github.com/openshift/oc/pkg/cli/gettoken/jwt"
+	"github.com/openshift/oc/pkg/cli/gettoken/oidc"
+	"github.com/openshift/oc/pkg/cli/gettoken/pkce"
+	"github.com/openshift/oc/pkg/cli/gettoken/tokencache"
+)
+
+var (
+	getTokenLong = templates.LongDesc(`
+	Experimental: This command is under development and may change without notice.
+	Built-in Credential Exec plugin of the oc.
+
+	It supports Auth Code, Auth Code + PKCE in addition to refresh token.
+	get-token caches the ID token and Refresh token after the auth code flow is
+	successfully completed and once ID token expires, command tries to get the
+	new token by using the refresh token flow. Although it is optional, command
+	also supports getting client secret to behave as an confidential client.
+`)
+	getTokenExample = templates.Examples(`
+	# Starts an auth code flow to the issuer url with the client id and the given extra scopes
+	oc get-token --client-id=client-id --issuer-url=test.issuer.url --extra-scopes=email,profile
+
+	# Starts an authe code flow to the issuer url with a different callback address and 5 minute authentication timeout.
+	oc get-token --client-id=client-id --issuer-url=test.issuer.url --callback-addresses=localhost:8343 --authentication-timeout=5m
+`)
+)
+
+const defaultCallbackAddress = "127.0.0.1:8000"
+
+type GetTokenOptions struct {
+	IssuerURL             string
+	ClientID              string
+	ClientSecret          string
+	ExtraScopes           []string
+	CallbackAdresses      []string
+	CACertFilename        string
+	InsecureTLS           bool
+	AuthenticationTimeout time.Duration
```

Comment:
yes definitely (of course with some default value 30 seconds). Because user may have very slow bandwidths and they may want to configure this value to authenticate.

Reply:
I believe you could just default to 5 minutes and remove this.

Reply:
Removed and set to 5 minutes.